modules/javafx.graphics/src/test/java/test/com/sun/javafx/css/StyleManagerTest.java

```diff
@@ -1120,4 +1122,53 @@ public void testCalculateCheckSum() {
 
         assertTrue(Arrays.equals(expectedChecksum, checksum));
     }
+
+    @Test
+    public void testSetSceneUserAgentStylesheetFromDataURI() {
+        var rect = new Rectangle();
+        var root = new StackPane(rect);
+        rect.getStyleClass().add("rect");
+
+        // Stylesheet content: .rect { -fx-fill: blue; }
+        Scene scene = new Scene(root);
+        scene.setUserAgentStylesheet("data:base64,LnJlY3QgeyAtZngtZmlsbDogYmx1ZTsgfQ==");
+        scene.getRoot().applyCss();
+
+        assertEquals(Color.BLUE, rect.getFill());
+    }
+
+    @Test
+    public void testSetSubSceneUserAgentStylesheetFromDataURI() {
+        var rect = new Rectangle();
+        var root = new StackPane(rect);
+        rect.getStyleClass().add("rect");
+
+        // Stylesheet content: .rect { -fx-fill: blue; }
+        var subScene = new SubScene(root, 100, 100);
+        subScene.setUserAgentStylesheet("data:base64,LnJlY3QgeyAtZngtZmlsbDogYmx1ZTsgfQ==");
+
+        Scene scene = new Scene(new StackPane(subScene));
+        scene.getRoot().applyCss();
+
+        assertEquals(Color.BLUE, rect.getFill());
+    }
+
+    @Test
+    public void testSetApplicationUserAgentStylesheetFromDataURI() {
+        var rect = new Rectangle();
+        var root = new StackPane(rect);
+        rect.getStyleClass().add("rect");
+        String uaStylesheet = Application.getUserAgentStylesheet();
+
+        try {
+            // Stylesheet content: .rect { -fx-fill: blue; }
+            Application.setUserAgentStylesheet("data:base64,LnJlY3QgeyAtZngtZmlsbDogYmx1ZTsgfQ==");
+            Scene scene = new Scene(root);
+            scene.getRoot().applyCss();
+
+            assertEquals(Color.BLUE, rect.getFill());
+        } finally {
+            Application.setUserAgentStylesheet(uaStylesheet);
```

Comment:
This line in `finally` block results in `ClassNotFoundException`

Run the tests as below to get the log-

`gradle --info :graphics:test &> test.log`

Reply:
Good catch. The reason for this is that the default Stylesheet is in the `javafx.controls` module, which is not available when running unit tests in the `javafx.graphics` module. I recommend to either `@Ignore` this test or else reset it to an empty style sheet (which is effectively what it will be for `javafx.graphics` tests, since the platform initialization code that sets up the default stylesheet is never called for these tests).

Reply:
I've decided to reset the UA stylesheet to an empty stylesheet.